etcdctl/ctlv3/command/global.go

```diff
@@ -268,3 +288,76 @@ func authCfgFromCmd(cmd *cobra.Command) *authCfg {
 
 	return &cfg
 }
+
+func insecureDiscoveryFromCmd(cmd *cobra.Command) bool {
+	discovery, err := cmd.Flags().GetBool("insecure-discovery")
+	if err != nil {
+		ExitWithError(ExitError, err)
+	}
+	return discovery
+}
+
+func discoveryCfgFromCmd(cmd *cobra.Command) *discoveryCfg {
+	var err error
+	var domainStr string
+	if domainStr, err = cmd.Flags().GetString("discovery-srv"); err != nil {
+		ExitWithError(ExitBadArgs, err)
+	} else if domainStr == "" && cmd.Flags().Changed("discovery-srv") {
+		ExitWithError(ExitBadArgs, errors.New("empty string is passed to --discovery-srv option"))
+	}
+
+	cfg := discoveryCfg{
+		domain:   domainStr,
+		insecure: insecureDiscoveryFromCmd(cmd),
+	}
+	return &cfg
+}
+
+func endpointsFromCmd(cmd *cobra.Command) ([]string, error) {
+	eps, err := endpointsFromFlagValue(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	// If domain discovery returns no endpoints, check endpoints flag
+	if len(eps) == 0 {
+		eps, err = cmd.Flags().GetStringSlice("endpoints")
+	}
+
+	return eps, err
+}
+
+func discoverer(domain string) ([]string, error) {
+	srvs, err := srv.GetClient("etcd-client", domain)
+	if err != nil {
+		return nil, err
+	}
+	return srvs.Endpoints, nil
+}
+
+func endpointsFromFlagValue(cmd *cobra.Command) ([]string, error) {
+	discoveryCfg := discoveryCfgFromCmd(cmd)
+
+	// If we still don't have domain discovery, return nothing
+	if discoveryCfg.domain == "" {
+		return []string{}, nil
+	}
+
+	eps, err := discoverer(discoveryCfg.domain)
```

Comment:
just inline `discoverer` and get rid of the function?
```go
srvs, err := srv.getClient("etcd-client", domain)
if err != nil {
    return nil, err
}
eps := srvs.Endpoints
```

Reply:
modified